RFC-0012-built-in-native-function-namespace-manager.md

```diff
@@ -0,0 +1,108 @@
+
+
+## [Register Native Sidecar Functions in default namespace]
+
+Proposers
+
+* Kevin Tang
+* Feilong Liu
+
+## [Related Issues]
+
+
+RFC for new SPI for SQL invoked functions
+https://github.com/prestodb/rfcs/pull/40/files
+
+## Summary
+
+Until we can do all the constant folding on the sidecar we need to be able to do constant folding in Java, even when getting the function list from the sidecar.  Some functions will be cpp only functions, and those we want to skip constant folding for, since those will fail.
+
+The goal is to be able to enable the sidecar while preserving the constant folding functionality for functions that have java implementation and SQL implementation. Once this is finished, sidecar can be enabled internally and inline_sql_functions can be fully enabled in C++ clusters. There is currently no support for constant folding in C++ functions, but this will be addressed in a followup design.
+
+
+## Background
+
+### [Requirements]
+
+* All functions with Java implementations get constant folded on the coordinator
+* All functions without Java implementations skip constant folding
+* Both functions with and without Java implementations get registered as "builtin", meaning that they don't need to be fully qualified to be used
+
+Breakdown by case
+* Function with only java implementation: constant fold, existing behavior
+* Function with only C++ implementation: no constant fold
+* Function with java and c++ implementation: constant fold, existing behavior
+* Function with only SQL implementation: use sql implementation, existing behavior
+* Function with SQL and C++ implementation: use c++ implementation, new behavior
+
+
+### Non-goals
+
+[Question]
+* Use presto.default namespace for every built in function or have separate namespace for native sidecar functions?
+
+[Answer]
+* If we choose to use presto.default. prefix for every function (built in functions from coordinator and functions from sidecar), then we would need to resolve naming collisions by only choosing one implementation when there exists a coordinator implementation and a worker implementation.
+
+* If we choose to have separate namespaces for built in functions from coordinator and functions from sidecar, then this would require extra changes on the worker code. Currently the worker code is using one prefix for every function, and this prefix is configurable. However, you cannot set a different prefix for different functions coming from the sidecar; they would all be the same prefix.
+
+* Since adding more worker changes adds extra complexity and confusion, we decided to use presto.default. prefix for every function from sidecar that is registered.
```

Comment:
But then how are you handling conflicts then ? I do see it depends on the code you have suggested below. But can you describe what the desired outcome is in a functional manner here ?